pydis_site/apps/content/resources/guides/pydis-guides/contributing/bot.md

```diff
@@ -2,207 +2,643 @@
+            </ul>
+        </div>
+    </div>
+</div>
+<br>
+
+#### With More Things Running Locally
+You can run additional services on the host. For the site, refer to the [site contributing guide](../site) to learn how to start it on the host, in which case you will need to change `urls.site` in `config.yml` to wherever the site is being hosted. This guide won't go over how to install and start the other dependencies on the host. If possible, prefer to start the services through Docker to replicate the production environment as much as possible.
+
+#### Start Only the Bot with Docker
+This method will start only the bot using Docker. The site has to have been started somehow beforehand.
+
+Start the bot using Docker Compose while inside the root of the project directory: `docker-compose up --no-deps bot`.
+
+### Development Tips
+Now that you have everything setup, it is finally time to make changes to the bot!
 
-The bot can run with or without Docker.
-When using Docker, the site, which is a prerequisite, can be automatically set up too.
-If you don't use Docker, you have to first follow [the site guide](../site/) to set it up yourself.
-The bot and site can be started using independent methods.
-For example, the site could run with Docker and the bot could run directly on your system (AKA the _host_) or vice versa.
+#### Working with Git
 
-## Run with Docker
+If you have not yet [read the contributing guidelines](../contributing-guidelines), now is a good time.
+Contributions that do not adhere to the guidelines may be rejected.
 
-The following sections describe how to start either the site, bot, or both using Docker.
-If you are not interested in using Docker, see [this page](../site/) for setting up the site and [this section](#run-on-the-host) for running the bot.
+Notably, version control of our projects is done using Git and Github.
+It can be intimidating at first, so feel free to ask for any help in the server.
 
-If you get any Docker related errors, reference the [Possible Issues](../docker#possible-issues) section of the Docker page.
+[**Click here to see the basic Git workflow when contributing to one of our projects.**](../working-with-git/)
 
-### Site and bot
+#### Running tests
 
-This method will start both the site and the bot using Docker.
+[This section](https://github.com/python-discord/bot/blob/main/tests/README.md#tools) of the README in the `tests` repository will explain how to run tests.
+The whole document explains how unittesting works, and how it fits in the context of our project.
 
-Start the containers using Docker Compose while inside the root of the project directory:
+#### Reloading parts of the bot
+If you make changes to a cog, you might not need to restart the entire bot for the changes to take effect. The command `!cog reload <cog_name>` (or `!c r <cog_name>` in short) re-imports the files associated with the cog.
```

Comment:
`!cog reload <cog_name>` is not correct, it's `!cogs reload <cog_name>`
... Either way I prefer `ext`: `!ext r <name>` so can we include that too? :D